pkg/fdsrv/fdsrv_test.go

```diff
@@ -0,0 +1,193 @@
+// Copyright 2022 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package fdsrv
+
+import (
+	"os"
+	"syscall"
+	"testing"
+	"time"
+)
+
+// Returns an fd of the read side of a pipe that has the value 'x' written in
+// it.
+func xPipe(t *testing.T) int {
+	var p [2]int
+	if err := syscall.Pipe(p[:]); err != nil {
+		t.Fatal("pipe:", err)
+	}
+	if _, err := syscall.Write(p[1], []byte("x")); err != nil {
+		t.Fatal("write:", err)
+	}
+	syscall.Close(p[1])
```

Comment:
so I realize it can not happen, but ... it might make sense to check this close for an error anyway. Every once in a while we check for a can't happen and ... it is happening :-)

Reply:
done